src/proof/nova.rs

```diff
@@ -237,22 +261,35 @@ impl<'a, F: CurveCycleEquipped, C: Coprocessor<F>> RecursiveSNARKTrait<F, C1LEM<
         steps: Vec<C1LEM<'a, F, C>>,
         store: &Store<F>,
     ) -> Result<Self, ProofError> {
-        assert!(!steps.is_empty());
-        assert_eq!(steps[0].arity(), z0.len());
         let debug = false;
-        let z0_primary = z0;
-        let z0_secondary = Self::z0_secondary();
+        let first_step = steps.first().expect("steps can't be empty");
+        assert_eq!(first_step.arity(), z0.len());
 
-        let circuit_secondary = TrivialCircuit::default();
+        let secondary_circuit = TrivialCircuit::default();
 
         let num_steps = steps.len();
-        tracing::debug!("steps.len: {num_steps}");
-
-        // produce a recursive SNARK
-        let mut recursive_snark: Option<RecursiveSNARK<E1<F>>> = None;
+        info!("proving {num_steps} steps");
+
+        let mut recursive_snark = RecursiveSNARK::new(
```

Comment:
Ah, yes, indeed. We may need to reproduce the original `unwrap_or_else`. One way to do this is to explicitly define a local closure `initial_recursive_snark: Fn() -> RecursiveSNARK` that produces this and invoke it inside the loop.